### ISSUE TITLE
keras can't save the model with customized loss function?

### ISSUE THREAD

**Issue by adamluo1995** [fchollet/keras#6356]
Hi everyone.
I realize this problem when i try to load model saved by keras.model.save_model(). it will raise an Error say ('Unknown loss function', ':bar1_loss'). bar1_loss function was write on the python script where i build and train the model, Of course the load_model() function can't recognize it. Similar problem occured when the model i try to load contains Lamba layer  whichs function use global name , the load_model() function can't recognize the global name too. Is it really a problem or it can be fixed in some way? 
Thanks :)

**Comment by alexklibisz** [comment]
Not 100% certain, but you might be able to use the `custom_objects` argument in the `load_model` function. See https://github.com/fchollet/keras/issues/3911. I know it works for custom metrics, not sure about the loss function though.

**Comment by datumbox** [comment]
I have recently used custom loss functions in Keras 2.x without a problem. As Alex said you need to provide the function using thte custom_objects argument in the load_model method. 

An alternative but very hacky workaround is once you finish training, you could recombile the model to use a standard cost function and then persist it. Loading it should not be a problem and so does making predictions. I would discourage you from doing that though. So if you try the custom_objects approach and you still get issues, post here a snippet that reproduces the problem and I'll have a look.
